✨ Feat: #80 커스텀 AlertPopup 컴포넌트 추가

Mark3891 opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #80

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

- ViewModifier 기반 CustomAlertModifier 구현
- title, message, confirm/cancel 텍스트 및 액션 전달 가능
- View extension으로 customAlert() 제공
- PreviewProvider를 통한 사용 예시 추가

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->

<img width="300" alt="image" src="https://github.com/user-attachments/assets/7150d87e-49ab-40ce-999c-aea7d18a83f1" />

---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인
- [x] iPhone 15, iOS 17.4 환경에서 정상 동작
- [ ] 다크모드 테스트는 이후 이슈로 분리 예정 (#000)

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- Preview로 만들어놓은 예제 참고해서 사용하시면 됩니다.
---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

Mark3891 committed:
✨ Feat: 커스텀 AlertPopup 컴포넌트 추가

✨ Feat: 커스텀 AlertPopup 컴포넌트 추가

Why: SwiftUI에서 재사용 가능한 경고 팝업 컴포넌트가 필요했음

How:
- isPresented 바인딩을 통해 표시 여부 제어
- 제목, 메시지, 확인/취소 버튼 텍스트 및 액션 전달 가능
- 반투명 배경 및 중앙 고정 팝업 디자인 구현
- Hex 기반 Color 초기화 확장 기능 포함
- PreviewProvider를 통한 UI 확인 가능

Tag: #ui #component #popup #swiftui #designsystem

RoastedJM commented:
Alert의 경우, 색상 지정 별도로 한 부분 없이 iOS 기본 컴포넌트를 사용하고 있습니다. 
Context Menu도 마찬가지라서, 해당 컴포넌트들 색상은 Hex사용 없이 별도 지정 안해도 될 것 같습니다.
Modal 배경색의 경우 Gray_white 값이 들어갑니다. (#ffffff)

<img width="976" height="622" alt="image" src="https://github.com/user-attachments/assets/8965c7fc-13b4-4204-b3a5-a6e4a0f03ae0" />

Mark3891 committed:
♻️ Refactor: AlertPopup를 SwiftUI .alert 기반 컴포넌트로 리팩토링

♻️ Refactor: AlertPopup를 SwiftUI .alert 기반 컴포넌트로 리팩토링

Why: 커스텀 팝업 대신 시스템 .alert를 활용해 접근성 및 유지보수성 향상

How:
- ViewModifier 기반 CustomAlertModifier 구현
- title, message, confirm/cancel 텍스트 및 액션 전달 가능
- View extension으로 customAlert() 제공
- PreviewProvider를 통한 사용 예시 추가

Tag: #refactor #alert #swiftui #ui #component

Mark3891 commented:
@RoastedJM 알려주셔서 감사해요 반영해서 .customAlert 만들었어요~~

freeskyES reviewed (approved):
👍고생하셨어요 우디!
